apps/sms/desktop-mock/navigator_moz_mobilemessage.js

```diff
@@ -1538,9 +1557,7 @@
   };
 
   MockNavigatormozMobileMessage.retrieveMMS = function(id) {
-    var request = {
-      error: null
-    };
+    var deferred = defer();
     var msgs = messagesDb.messages;
     var idx = 0, len = msgs.length;
     setTimeout(function() {
```

Comment:
we don't need the setTimeout if we use promises -- it was here to make the callback call asynchronous but with promises we have this for free